### ISSUE TITLE
Requests for functionality that can display images containing tag A or images containing tag B (not just tags, but more elements) at the same time

### ISSUE THREAD

**Issue by NRZ091** [deckerst/aves#1320]
●→**Is your feature request related to a problem? Please describe.**

○→Under normal circumstances, you can select multiple labels in the label list and use the "Show in Collection" function to display the combination of multiple labels, but this can only display the same type (such as both labels/both albums, etc.). It is not possible to display different types at the same time (such as one album and one label).

I was trying to query the photo tag→"character:firefly" or time→" July 20 "and found that I couldn't do it

●→**Describe the solution you'd like**

○→I want to optimize the implementation of intersections, union sets, and complement sets between elements (not just tags)

If possible, it is best to mark the relationship between filters directly through the search box (if not, it does not matter).

●→Finally, I wish your software will get better and better. aves has provided me with great help in sorting photos on my mobile phone

**Comment by NRZ091** [comment]
Original text：

●→Title
○→对于能够实现同时展示包含标签A的图片或者包含标签B的图片(不仅仅是标签，更偏向元素)的功能的请求

●→Is your feature request related to a problem? Please describe.
○→正常情况下可以在标签列表中选择多标签后使用"Show in Collection"功能进行多标签的组合展示，但这只能展示同一类型(如二者均为标签/二者均为相册等等)没法实现不同类的同时展示(如一个为相册一个为标签时)

我刚才打算查询tag→"character:firefly"或time→"7月20日"的照片，然后发现没法进行这样的操作

●→Describe the solution you'd like
○→希望能够优化一下标记(不只是标签)之间的交集，并集，补集之间的实现

可以的话最好能直接通过搜索框进行标记筛选之间的关系处理(不可以的话也没关系)

●→Finally
○→最后祝你们的软件能够越来越好，aves对我手机端照片分类整理提供了巨大的帮助

**Comment by NRZ091** [comment]
> > I was trying to query the photo tag→"character:firefly" or time→" July 20 "
> 
> I understand your request, but I'd like to see a convincing example. The one you provided seems very strange to me: `tag or day`. Is that a real query?


This is indeed a relatively rare problem

(I have sorted out the picture of "折紙ぎょくお" shown below yesterday, so there are no duplicate pictures)

Yesterday I downloaded some new illustrations from the artist "折紙ぎょくお" from pixiv, and since I've downloaded his illustrations before, I wanted to confirm which ones I've already downloaded (since I've only recently started using aves, there are a lot of old pictures without tags, which are now being sorted out). Yesterday, I added the label "作者:折紙ぎょくお" to the new illustration, but the previous illustration did not add the label)
![Screenshot_2024-11-29-15-47-19-499_deckers thibault aves](https://github.com/user-attachments/assets/a54ac3c8-9c20-45b3-9b85-2163c124eaf3)

So I found one of his illustrations that I downloaded at that time (its date is July 20, 2024), and I selected the corresponding time label of this picture (July 20, 2024) from the elements of this photo, and then I wanted to add the label "作者:折紙ぎょくお" through the search box to realize the union display operation. To determine which of his illustrations I downloaded repeatedly
![Screenshot_2024-11-29-15-48-56-295_deckers thibault aves](https://github.com/user-attachments/assets/f4132b69-bcc2-46b2-ac00-4256b3639424)
![Screenshot_2024-11-29-15-49-08-154_deckers thibault aves](https://github.com/user-attachments/assets/295baa85-3b25-43a7-b495-e54a488b3aa0)

And then I realized that I couldn't do it (it seemed like I could only do the intersection display through the search box)
![Screenshot_2024-11-29-15-59-32-562_deckers thibault aves](https://github.com/user-attachments/assets/92da0fb1-c3a4-4d89-be86-adace33870c6)

Later, I found that it is possible to combine tags in media sets (but only with the same element type, as mentioned yesterday, both albums/both tags).


Follow-up solution: I selected his illustrations from the pictures on July 20 manually and added the label "作者:折紙ぎょくお". Then I selected the label" 作者:折紙ぎょくお" separately, sorted them according to size, and deleted the duplicate pictures


It is also hoped that the sorting method with the label selector can be temporary and does not affect the sorting of the normal media set.
Because I find it good to sort duplicate images by size, but my normal media set is sorted by time, so I have to change the normal media set from size sort to time sort after each collation)
→(Maybe this suggestion should be re-opened as an issue, but it doesn't matter if this is not changed)

Original text：
这确实是一个比较少见的问题

(以下展示的"折紙ぎょくお"的图片我昨天已经整理过了，所以并没有出现重复的图片)

昨天我从pixiv上下载了画师"折紙ぎょくお"的一些新插画，然后因为我之前也下载过他的插画作品，想要确认一下哪些是我已经下载过的(因为我是最近才开始使用aves的，所以有大量之前的图片并没有添加标签，现在正在进行整理中，昨天时我新下的插画是有添加"作者:折紙ぎょくお"标签的，但是之前下的他的图并没有添加标签)

于是我找到了当时下载过的他的插画的其中一张(其时间为2024年7月20日)，并且我从这张照片的元素中选择了这张图片对应的时间标签(2024年7月20日)，然后我想通过搜索框添加"作者:折紙ぎょくお"的标签实现并集展示操作，来确定哪些是我重复下载的他的插图

然后我发现我实现不了(貌似通过搜索框只能实现交集展示)

之后我发现在媒体集中倒是可以实现标签的并集展示(但是只能局限与相同的元素类型，如昨天提到的两者同为相册/两者同为标签)


后续解决方法：我手动在7月20日的图片中筛选出他的插画并添加"作者:折紙ぎょくお"标签，后续单独选择"作者:折紙ぎょくお"标签，按照大小排序，并删除重复的图片


(另外希望有标签选择器时的排序方式能是临时的，不去影响正常媒体集的排序。
因为我发现通过大小排序来确认重复图片效果不错，但是我正常媒体集的使用是按照时间排序，所以我每次整理之后都要把正常媒体集从大小排序再换回时间排序)
→(或许这条建议要重新开一条issue，不过这个不改其实也没关系)

**Comment by deckerst** [comment]
> Follow-up solution: I selected his illustrations from the pictures on July 20 manually and added the label "作者:折紙ぎょくお". Then I selected the label" 作者:折紙ぎょくお" separately, sorted them according to size, and deleted the duplicate pictures

Indeed, that's the easiest way to do it.

> It is also hoped that the sorting method with the label selector can be temporary and does not affect the sorting of the normal media set.

Related:
- #207

**Comment by NRZ091** [comment]
> > Follow-up solution: I selected his illustrations from the pictures on July 20 manually and added the label "作者:折紙ぎょくお". Then I selected the label" 作者:折紙ぎょくお" separately, sorted them according to size, and deleted the duplicate pictures
> 
> Indeed, that's the easiest way to do it.
> 
> > It is also hoped that the sorting method with the label selector can be temporary and does not affect the sorting of the normal media set.
> 
> Related:
> 
> * [Sort/Group type per individual album #207](https://github.com/deckerst/aves/issues/207)

Oh,this is also a really nice feature too

In fact, I am not referring to the separate classification between albums, but the following situation (and this actually does not have a big impact, at most, it is a bit more to modify the sorting method and then change back, so it does not matter if you do not adjust this)

Finally, thank you for your hard work

**Comment by deckerst** [comment]
@gagarwa for tags, you can select multiple tags on the `Tags` page, then select `Show in Collection` in the top right menu to show their union.

**Comment by gagarwa** [comment]
> [@gagarwa](https://github.com/gagarwa) for tags, you can select multiple tags on the `Tags` page, then select `Show in Collection` in the top right menu to show their union.

Yes, I'm aware of this, but I have 500+ tags, so manually selecting them is difficult.
